homeassistant/components/myuplink/number.py

```diff
@@ -0,0 +1,132 @@
+"""Number entity for myUplink."""
+
+
+from aiohttp import ClientError
+from myuplink import DevicePoint
+
+from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import MyUplinkDataCoordinator
+from .const import DOMAIN
+from .entity import MyUplinkEntity
+from .helpers import find_matching_platform
+
+DEVICE_POINT_UNIT_DESCRIPTIONS: dict[str, NumberEntityDescription] = {
+    "DM": NumberEntityDescription(
+        key="degree_minutes",
+        icon="mdi:thermometer-lines",
+        native_unit_of_measurement="DM",
+    ),
+}
+
+CATEGORY_BASED_DESCRIPTIONS: dict[str, dict[str, NumberEntityDescription]] = {
```

Comment:
Why isn't it sufficient with the first one as they both result in the same entity description?

Reply:
The world is not perfect...
Or the upstream implementation of the API is not perfect.
The myUplink  cloud service is an attempt from the Nibe group to consolidate the service for all their brands (Nibe, CTC, Contura, Ceteterm and more). Most of these heatpumps report DM as unit_of_meaurement for degree-minutes data while Nobe F-series products leave this field empty. But all F-series appliances feed this data point from the same parameter_id. That is why we need an override table, also used in other platforms in this integration.